packages/dataviews/src/components/dataviews-pagination/index.tsx

```diff
@@ -20,13 +25,25 @@ function DataViewsPagination() {
 		view,
 		onChangeView,
 		paginationInfo: { totalItems = 0, totalPages },
+		data,
 	} = useContext( DataViewsContext );
 
+	const currentPage = view.page ?? 1;
+
+	// If the current page is not the first page, and there are no items on the current page, go to the previous page.
+	useEffect( () => {
+		if ( currentPage !== 1 && ! data.length ) {
+			onChangeView( {
+				...view,
+				page: currentPage - 1,
+			} );
+		}
+	}, [ data.length, currentPage, view, onChangeView ] );
```

Comment:
This effect isn't reliable. What happens when you're not on the first page and data is fetching? The condition will be `true`, and the user will be redirected to the previous page.

P.S. I know the current page state doesn't persist in the URL, but that will change.